test/parallel/test-crypto-dh.js

```diff
@@ -86,8 +86,8 @@ const crypto = require('crypto');
   }
 
   {
-    const v = crypto.constants.OPENSSL_VERSION_NUMBER;
-    const hasOpenSSL3WithNewErrorMessage = (v >= 0x300000c0 && v <= 0x30100000) || (v >= 0x30100040 && v <= 0x30200000);
+    const hasOpenSSL3WithNewErrorMessage = (common.hasOpenSSL(3, 0, 12) && !common.hasOpenSSL(3, 1, 0)) ||
+                                           (common.hasOpenSSL(3, 1, 4) && !common.hasOpenSSL(3, 2, 0));
```

Comment:
I think the new check fails for versions 3.1.0 and 3.2.0. `!common.hasOpenSSL(3, 1, 0)` returns `false`, no?

Reply:
🤦 You're right, I'm off by one. Will update.

Reply:
hmm actually the check is expected to fail for 3.1.0? The check is from https://github.com/nodejs/node/pull/50395:
> Error message change is test-only and uses the right error message for versions >= 3.0.12 in 3.0.x and >= 3.1.4 in 3.1.x series.

So 3.1.0 should be the old message (i.e. check should fail).

Reply:
I'm not sure but the original `hasOpenSSL3WithNewErrorMessage` is `true` for 3.1.0. Now is `false`.

Reply:
I'll change it to match what the test was doing before but I'm not convinced it's correct..